atc/engine/engine.go

```diff
@@ -214,7 +224,7 @@ func (b *engineBuild) Run(logger lager.Logger) {
 
 	done := make(chan error)
 	go func() {
-		ctx := lagerctx.NewContext(b.ctx, logger)
+		ctx = lagerctx.NewContext(ctx, logger)
```

Comment:
Reassigning `ctx` in a goroutine here seems like it could be a race condition waiting to happen. :thinking: It's fine right now because nothing uses `ctx` after the goroutine but this seems like it might be hard to catch in the future. I'm not sure if we build and run Concourse with `--race` anywhere in the pipeline.

Reply:
oh, we definitely don't need the definition there

>  I'm not sure if we build and run Concourse with --race anywhere in the pipeline.

I remember chatting about this with someone (tried searching the place, but found nothing lol) that we could perhaps have `smoke` running with `-race` (although, I think we'd need to build a version of `concourse` with `-race` as that's a build-time modification)